### PR TITLE
feat: support multiple model binaries

### PR DESCRIPTION
**Pull request by louis-jan**
## Describe Your Changes
This is the decoupled PR from #1506 to support multiple model binaries. Detach from #1506 to unblock model linking cc @namchuai 
```json
{
   "source":  {
        "url": "https://huggingface.../tinyllama.gguf",
        "filename": "tinyllama.gguf" // Optional
   }
}
```

NOTICE:
- There is models migration step will be applied when user update their app. Its to convert old model.json to latest syntax.
## Related Issues

- #1381

## Self Checklist

- [x] Added relevant comments, esp in complex areas
- [x] Updated docs (for bug fixes / features)
- [ ] Created issues for follow-up changes or refactoring needed

### CODE REVIEW

**Comment by louis-jan** [comment]
> @namchuai @louis-jan The function for `downloadModel` at https://github.com/janhq/jan/blob/main/core/src/node/api/common/builder.ts#L249 seems out of date and duplicated with the code in `electron/handlers/download.ts` and `extensions/model-extension/src/index.ts`
> 
> However, currently only `extensions/model-extension/src/index.ts` is up to update to support multiple binaries

@hiro-v Great find, this is a part of upcoming refactor, it's to share these logics between node module & electron handlers.

**Review by namchuai — APPROVED** [review]

**Comment by louis-jan** [comment]
@hahuyhoang411 Please help review all the model.json changes. Thanks!

File: models/dolphin-2.7-mixtral-8x7b/model.json
Comment (hahuyhoang411) [inline models/dolphin-2.7-mixtral-8x7b/model.json:1]: Are we missing `llama_model_path` for this model?

```
"llama_model_path": "dolphin-2.7-mixtral-8x7b.Q4_K_M.gguf"
```

**Comment by hahuyhoang411** [comment]
I reviewed the PR and added missing fields.

cc @louis-jan

**Comment by louis-jan** [comment]
> I reviewed the PR and added missing fields.
> 
> 
> 
> cc @louis-jan 

Nice! Thanks.

**Review by hahuyhoang411 — APPROVED** [review]

**Pull request merged**